### PR TITLE
Refactor RPM unassociation tests

### PR DESCRIPTION
**Pull request by Ichimonji10**
Do at least the following:

* Rename class `RemoveAssociatedUnits` to `RemoveUnitsTestCase`.
* Update the assertions on `RemoveUnitsTestCase`. Drop the assertion
  that checks that a unit randomly selected from a list is present in
  the list. Of course it is - that's guaranteed by Python itself.
  Refactor the remaining assertions to, hopefully, provide more concise
  and useful messages upon test failure.
* Heavily refactor the private methods in the module, and make use of
  them as appropriate.

This commit does not introduce any failures. Test results generated
with:

    python -m unittest2 pulp_smash.tests.rpm.api_v2.test_unassociate

### CODE REVIEW

**Comment by Ichimonji10** [comment]
For the record, this refactor was basically a chance for me to experiment with Pulp's [criteria](http://pulp.readthedocs.org/en/latest/user-guide/general-reference.html#criteria).

**Comment by Ichimonji10** [comment]
As it happens, this change will also fix a bug in the test suite. In the case where two or more RPMs have the same name (such as with the "walrus" RPM, for which there are two versions), `pulp_smash.tests.rpm.api_v2.test_unassociate.RemoveAssociatedUnits.test_one_unit_removed` currently fails.

File: pulp_smash/tests/rpm/api_v2/test_unassociate.py [hunk 1]
```diff
@@ -92,50 +100,28 @@ def setUpClass(cls):
...
+            unit = random.choice(_get_units_by_type(cls.units_before, type_id))
+            cls.units_removed.append(unit)
+            _remove_unit(cls.cfg, repo['_href'], unit)
+        cls.units_after = _search_units(cls.cfg, repo['_href'], cls.TYPE_IDS)
+
+    def test_units_not_in_search(self):
+        """Assert the removed units do not appear in search results."""
+        ids_removed = {_get_unit_id(unit) for unit in self.units_removed}
+        ids_after = {_get_unit_id(unit) for unit in self.units_after}
+        self.assertEqual(ids_removed & ids_after, set())  # s1.isdisjoint(s2)
```
Comment (elyezer) [inline pulp_smash/tests/rpm/api_v2/test_unassociate.py]: Smart move, using assertTrue and isdisjoint will give the same result, but using assertEqual with the two sets will actually prints what are the differences. :100:

**Comment by elyezer** [comment]
All this code makes sense and are good to me.

Feel free to merge it.

File: pulp_smash/tests/rpm/api_v2/test_unassociate.py (hunk 1 above)
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_unassociate.py]: :smile:

**Comment by Ichimonji10** [comment]
Thanks for looking this over, @elyezer. This still looks good to me. Merging.

**Pull request merged**